app/notification.go

```diff
@@ -474,9 +483,10 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 			}
 			if sendEvent {
 				message := model.NewWebSocketEvent(model.WebsocketEventThreadUpdated, team.Id, "", uid, nil)
-				threadMembership, err := a.Srv().Store.Thread().GetMembershipForUser(uid, post.RootId)
-				if err != nil {
-					return nil, errors.Wrapf(err, "cannot get thread membership %q for user %q", post.RootId, uid)
+				threadMembership := participantMemberships[uid]
+				if threadMembership == nil {
+					a.Log().Warn("Missing thread membership for participant in notifications.", mlog.String("user_id", uid), mlog.String("thread_id", post.RootId))
+					continue
```

Comment:
`participantMemberships` will only contain members who have been mentioned in the post. The loop here notifies all followers of the thread. A person who manually followed the thread wont be mentioned (or be a participant), so the won't get notified here. I think for `threadMembership == nil` we need to query the DB for that membership instead of just logging the error.